File: client-app/src/App/modules/Podrucje.ts
import { Biljka } from "./Biljke";
import { Koordinata } from "./Koordinata";

export interface Podrucje{
    id: string;
    oblast: string;
    koordinate: Koordinata[];
    biljke: Biljka[];
}

export class Podrucje implements Podrucje{
    constructor(init?:PodrucjeFormValue){
        Object.assign(this,init);
    }
}

export class PodrucjeFormValue{
    id: string = '';
    oblast: string = '';
    koordinate: Koordinata[] = [];
    biljke: Biljka[] = [];

    constructor(podrucje?:PodrucjeFormValue){
        if(podrucje){
            this.id = podrucje.id;
            this.oblast = podrucje.oblast;
            this.koordinate = podrucje.koordinate;
            this.biljke = podrucje.biljke;
        }
    }
}

File: Domain/src/Podrucje.cs
using System.Text.Json.Serialization;

namespace Domain;

public class Podrucje
{
    public Guid ID { get; set; }
    public string Oblast { get; set; }
    public ICollection<KoordinatePodrucja> Koordinate { get; set; } = new List<KoordinatePodrucja>();
    //[JsonIgnore]
    public ICollection<Podrucja_Biljke> Biljke { get; set; } = new List<Podrucja_Biljke>();
    [JsonIgnore]
    public ICollection<Obilazak> Obilasci { get; set; } = new List<Obilazak>();
    public ICollection<Ruta> Rute {get; set; } = [];

}